File: backend/app/schemas/suggestion.py
from __future__ import annotations

from pydantic import BaseModel


class SuggestionResponse(BaseModel):
    id: str
    project_id: str
    source_run_id: str | None
    provider: str
    config_diff: str
    rationale: str
    evidence_json: str | None
    expected_effect: str | None
    tradeoffs: str | None
    confidence: float | None
    risk_level: str | None
    status: str
    applied_config_version_id: str | None
    created_at: str
    resolved_at: str | None

    model_config = {"from_attributes": True}


class SuggestionListResponse(BaseModel):
    items: list[SuggestionResponse]
    total: int


class SuggestionGenerateRequest(BaseModel):
    source_run_id: str | None = None
    notes: str | None = None


class SuggestionResolve(BaseModel):
    action: str

File: frontend/src/types/suggestion.ts
import type { ConfigDiff } from "./config";

export type SuggestionStatus = "pending" | "accepted" | "rejected" | "applied" | "expired";
export type SuggestionProvider = "anthropic" | "openai_compatible" | "rule_engine";
export type RiskLevel = "low" | "medium" | "high";

export interface SuggestionEvidence {
  readonly type: string;
  readonly referenceId: string;
  readonly label: string;
  readonly value: string | number;
}

export interface AISuggestion {
  readonly id: string;
  readonly projectId: string;
  readonly sourceRunId: string | null;
  readonly provider: SuggestionProvider;
  readonly configDiff: ConfigDiff;
  readonly rationale: string;
  readonly evidence: ReadonlyArray<SuggestionEvidence>;
  readonly expectedEffect: string | null;
  readonly tradeoffs: string | null;
  readonly confidence: number | null;
  readonly riskLevel: RiskLevel | null;
  readonly status: SuggestionStatus;
  readonly appliedConfigVersionId: string | null;
  readonly createdAt: string;
  readonly resolvedAt: string | null;
}
